Update some visual details in the add-template modal(s)

jameskoster opened the pull request:
## What?
Part of https://github.com/WordPress/gutenberg/issues/50134.

Updates the appearance of the add-template modal contents to be consistent with itself, and other parts of the UI.

In the first step, the buttons are now more closely aligned to similar button styles, IE any `tertiary` button variants, and blocks in the Inserter.

The list items in the second step now resemble the suggestions in the Command Center.

## How?
Mostly css. Plus a couple of instances of the `Text` component. 

Side note: We should update the `Text` component to include the correct line-height on the `body` size.

## Testing Instructions
1. Add a template that requires further scope selection, e.g. Single Post
2. Observe the new styling / hover treatment
3. Select "Single item"
4. Observe the new styling / hover treatment

### Before

https://user-images.githubusercontent.com/846565/234905649-5e8aff2c-134e-4b9c-bdea-8aaad1a7f769.mp4

### After
https://user-images.githubusercontent.com/846565/234905627-66f6cf6d-3bd8-40c5-8560-ef1791f35d99.mp4

jameskoster committed:
Buttons and modal dimensions

jameskoster committed:
List items and other styles

jasmussen commented:
Thanks for all the attention to detail here. Just connecting the dots with [this border feedback](https://github.com/WordPress/gutenberg/issues/50134#issuecomment-1525958968) I gave here, it seems okay to keep the gray borders in the resting state, perhaps make them lighter, similar to style buttons. And then if it still works, just omit the borders from the hover state.

jameskoster commented:
Copy pasting my reply from the issue :)

> I kind of like removing them, to be consistent with Inserter buttons, etc.

jameskoster commented:
@jasmussen how do you feel about this?


https://user-images.githubusercontent.com/846565/235665011-171f43c6-ad48-49ab-b753-39e4c95017b1.mp4

jameskoster committed:
spit scope selector into rows

jasmussen commented:
I liked the big buttons, but acknowledge they only work with the gray borders, which I still like. Big tappable buttons, even in their resting state, not just on hover. It's not the strongest opinion in the world, and I do appreciate that the hover state of the list-items now line up. But it might still be good to look at this button type in the larger range of things we want to do with page and template modals, it might work converting it all to lists here, but we might still need the larger buttons elsewhere. 

Normally I agree the borders add noise, in this case, I think it's about managing the whitespace.

jameskoster committed:
Revert "spit scope selector into rows"

This reverts commit 6f4cc2c43ef42e242981c14d2833088fe7ec268d.

jameskoster committed:
Add borders

jameskoster commented:
<img width="522" alt="Screenshot 2023-05-02 at 17 03 26" src="https://user-images.githubusercontent.com/846565/235721676-8668fd1a-62f2-44f7-8f19-a0a2ec94c785.png">

Okay let's disagree and commit 😁

jasmussen reviewed (approved):
Sorry about the back and forth. I think this looks great:

<img width="634" alt="Screenshot 2023-05-03 at 10 09 23" src="https://user-images.githubusercontent.com/1204802/235863073-17cfc409-0327-4b9c-87f9-838822194e35.png">
<img width="619" alt="Screenshot 2023-05-03 at 10 09 27" src="https://user-images.githubusercontent.com/1204802/235863077-3420ea6b-7845-429d-a9cb-b6636da4241b.png">

jameskoster committed:
Alignment in suggestions list

jameskoster commented:
Ooops, the revert above removed the negative margins on the suggestions list that I added to line things up in that view. I just added those back:

<img width="485" alt="Screenshot 2023-05-03 at 10 06 40" src="https://user-images.githubusercontent.com/846565/235875839-34f89b34-107e-4749-bb81-46a608a708aa.png">

I'll merge when the checks pass.
